src/pages/UserPage.jsx

```diff
@@ -0,0 +1,427 @@
+import Dynamicform from "../components/dynamicform";
+import CustomButton from "../components/button";
+import Tooltip from "../components/toolTip";
+import WithLayoutComponent from "../hocs/WithLayoutComponent";
+import Toast from "../components/toast";
+import IssuanceForm from "../components/issuanceform";
+import Loader from "../components/loader";
+
+import back from "../assets/images/go-back.png";
+import next from "../assets/images/go-next.png";
+import EditIcon from "../assets/images/editicon.png";
+import DeleteIcon from "../assets/images/deleteicon.png";
+import assign from "../assets/images/bookaddd.png";
+import historyicon from "../assets/images/historyicon.png";
+
+import { useNavigate } from "react-router-dom";
+import { modalSizes } from "../components/utils";
+
+function UsersPage() {
+  const [users, setUsers] = useState([]);
+  const [books, setBooks] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [modalType, setModalType] = useState("");
+  const [currentData, setCurrentData] = useState({});
+  const [currentPage, setCurrentPage] = useState(0);
+  const [pageSize] = useState(7);
+  const [totalPages, setTotalPages] = useState(0);
+  const [searchTerm, setSearchTerm] = useState("");
+  const [selectedBook, setSelectedBook] = useState(null);
+  const [issueDate, setIssueDate] = useState("");
+  const [returnDate, setReturnDate] = useState("");
+  const [status, setStatus] = useState("issued");
+  const [issuanceType, setIssuanceType] = useState("");
+
+  const [showToast, setShowToast] = useState(false);
+  const [toastType, setToastType] = useState("success");
+  const [toastMessage, setToastMessage] = useState("");
+  const showSuccessToast = (message) => {
+    setToastType("success");
+    setToastMessage(message);
+    setShowToast(true);
+  };
+  const showFailureToast = (message) => {
+    setToastType("failure");
+    setToastMessage(message);
+    setShowToast(true);
+  };
+
+  const navigate = useNavigate();
+
+  const getUsers = async () => {
+    const trimmedSearchTerm = searchTerm.trim();
+
+    if (trimmedSearchTerm.length >= 3 || trimmedSearchTerm.length === 0) {
+      try {
+        setLoading(true);
+
+        const data = await fetchUsers(currentPage, pageSize, trimmedSearchTerm);
+        setUsers(data.content || []);
+        setTotalPages(data.totalPages || 0);
+      } catch (error) {
+        console.error("Error fetching users:", error);
+      } finally {
+        setLoading(false);
+      }
+    } else if (trimmedSearchTerm.length > 0 && trimmedSearchTerm.length < 3) {
+    } else {
+      setUsers([]);
+      setTotalPages(0);
+    }
+  };
+
+  useEffect(() => {
+    getUsers();
+  }, [currentPage, searchTerm]);
+
+  const getBooks = async () => {
+    try {
+      const data = await fetchAllBooks();
+      setBooks(data || []);
+    } catch (error) {
+      showFailureToast("Can't fetch Books")
```

Comment:
don't use hard coded message, use message coming from the backend.

Reply:
yes.did it already in the new push